### PR TITLE
fix crash on x % 0`

### PR DESCRIPTION
**Pull request by tbg**
found via go-fuzz

ping @dvyukov for your trophy list.

### CODE REVIEW

**Comment by dvyukov** [comment]
added to trophy list

File: sql/parser/eval.go [hunk 1]
```diff
@@ -231,6 +232,10 @@ var binOps = map[binArgs]func(Datum, Datum) (Datum, error){
 	},
 
 	binArgs{Mod, intType, intType}: func(left Datum, right Datum) (Datum, error) {
+		r := right.(DInt)
```
Comment (tamird) [inline sql/parser/eval.go]: nit: no need for this local

**Comment by tamird** [comment]
LGTM

File: sql/parser/eval.go (hunk 1 above)
Comment (tschottdorf) [inline sql/parser/eval.go]: whoops, meant to use it below instead of `right.(DInt)`. Will fix.

Comment (tamird) [inline sql/parser/eval.go]: :+1:

**Comment by petermattis** [comment]
LGTM

See #1622.

**Pull request merged**